.github/workflows/integration_tests.yml

```diff
@@ -773,15 +820,43 @@ jobs:
           fi
 
   test_desktop:
-    name: test-desktop-${{ matrix.os }}-${{ matrix.ssl_variant }}
+    name: test-desktop-${{ matrix.os }}-${{ matrix.arch }}-${{ matrix.ssl_variant }}
     needs: [check_and_prepare, build_desktop]
     runs-on: ${{ matrix.os }}
     if: contains(needs.check_and_prepare.outputs.matrix_platform, 'Desktop') && needs.check_and_prepare.outputs.apis != '' && !cancelled()
     strategy:
       fail-fast: false
       matrix:
         os: ${{ fromJson(needs.check_and_prepare.outputs.matrix_os) }}
+        arch: ${{ fromJson(needs.check_and_prepare.outputs.matrix_arch_combined) }}
         ssl_variant: ${{ fromJson(needs.check_and_prepare.outputs.matrix_ssl) }}
+        # Because matrix_arch_combined combines Mac, Linux, and Windows, we need
+        # a few exclusions. These are standard exclusions used in several
+        # places.
+        exclude:
+        # Do not attempt to use arm64 on Windows or Linux.
+        - os: windows-latest
+          arch: arm64
+        - os: ubuntu-latest
+          arch: arm64
+        # Do not attempt to use x86 on Mac.
+        - os: macos-12
+          arch: x86
+        # Until we support building openssl from source, we can't use the
+        # system's openssl when cross-compiling, except on Linux. Builds all
+        # happen on x64 machines, so arm64 and x86 are technically
+        # cross-compiling.
+        - os: windows-latest
+          ssl_variant: openssl
+          arch: x86
+        - os: macos-12
```

Comment:
Is this case also included in the following one?

Reply:
Yep, it is. I I kept the first one in just to be consistent between the two sets of exclusions (for build and for test) but I can just remove the redundant one.